tests/bug-reports/Bug1355.fst

```diff
@@ -0,0 +1,87 @@
+module Bug1355
+
+(***** Example 1 from the issue, still fails *****)
+
+assume val t: v:(int -> Type) -> (i:int -> v i -> Type0) -> Type
+
+irreducible type w #v #r (x:t v r) = True
+
+#push-options "--__temp_no_proj  Bug1355"
```

Comment:
fails because of refinement, we discussed addressing it by disabling SMT for projectors and using more aggressive normalization

Reply:
This is fixed now (via the equality ascription).